src/data/nat/periodic.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2021 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+import algebra.periodic
+import data.nat.interval
+import data.nat.count
+
+/-!
+# Periodic Functions on ℕ
+
+This file identifies a few functions on `ℕ` which are periodic, and also proves a lemma about
+periodic predicates which helps determine their cardinality when filtering intervals over them.
+-/
+
+open finset nat function
+
+lemma periodic_gcd (a : ℕ) : periodic (gcd a) a :=
```

Comment:
All these results should be in the `nat` namespace, because they are not about the generic `gcd` etc... but only about the `nat`-specific one.

Reply:
Thanks for fixing this in your branch.